event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/resource/EventPageResource.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.services.eventsourcing.source.api.resource;
+
+
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.BACKWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.FORWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.valueOf;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.FIRST;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.HEAD;
+
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.Page;
+import uk.gov.justice.services.eventsourcing.source.api.security.AccessController;
+import uk.gov.justice.services.eventsourcing.source.api.service.EventsPageService;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.sql.SQLException;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.UriInfo;
+
+@Path("event-streams/{streamId}/{position}/{direction}/{pageSize}")
+public class EventPageResource {
+
+    @Context
+    HttpHeaders headers;
+
+    @Inject
+    private EventsPageService eventsPageService;
+
+    @Inject
+    private AccessController accessController;
+
+    @GET
+    @Produces("application/vnd.event-source.events+json")
+    public Page events(
+            @PathParam("streamId") final String streamId,
+            @PathParam("position") final String position,
+            @PathParam("direction") final String direction,
+            @PathParam("pageSize") final long pageSize,
+            @Context final UriInfo uriInfo) throws URISyntaxException, SQLException, MalformedURLException {
+
+        if (HEAD.getPosition().equals(position) && FORWARD.toString().equals(direction)) {
```

Comment:
I think the errors might need to be updated? to be clearer...

Reply:
Backward and Forward are directions of the query and Previous and Next are the page link directions. Errors are currently for page links so i think its fine

Reply:
sure